### DISCUSSION TITLE
Git backend and CLI usage

### DISCUSSION THREAD

**Discussion by cannadayr** [emad-elsaid/xlog#65] [Ideas]
Hello,

For context, I am not a current user of xlog, I'm evaluating personal wiki software and xlog (and ikiwiki) have both stood out to me.

Currently, new revisions to files are hashed and their contents added to the filename.md.versions/ directory.
My workflow is often in the terminal, where updates to a file would not be hashed and added to the versions directory.
I have seen other wiki's use git as a backend, so instead of a versions directory, it would create a git commit.
This would enable people to use the wiki from either the command line or the web interface, and make it easier to use their existing git workflow to perform backups.

Has this idea been considered? There might need to be additional discussion if this should apply to binary media files.

**Comment by emad-elsaid** [comment]
* Versioning with Git instead of writing `/.versions/` is an idea that crossed my mind, especially that I use git for my knowledgebase 
* one thing that stopped me from doing that is that I'm not sure it's a good idea to commit to Git in the background with every save. it feels like committing your code file to git with every save. 
* However keeping versions is implemented using [an extension](https://github.com/emad-elsaid/xlog/blob/master/extensions/versions/versions.go)  
* So technically the behavior can be switched to commit the file to git. should be straightforward [this function](https://github.com/emad-elsaid/xlog/blob/master/extensions/versions/versions.go#L19) should invoke Git for the passed page filename
*

**Comment by cannadayr** [comment]
Interesting! Thank you. If I end up implementing [something] that lets you commit changes from the web ui I'll let you know. By the way, great tool I enjoy how minimal and out-of-the-way it is.
